src/Workspaces/Core/Portable/Rename/RenameLocation.ReferenceProcessing.cs

```diff
@@ -537,10 +546,51 @@ private static Regex GetRegexForMatch(string matchText)
                 return new Regex(matchString, RegexOptions.CultureInvariant);
             }
 
-            internal static string ReplaceMatchingSubStrings(string replaceInsideString, string matchText, string replacementText)
+            internal static string ReplaceMatchingSubStrings(
+                string replaceInsideString,
+                string matchText,
+                string replacementText,
+                ImmutableSortedSet<TextSpan>? subSpansToReplace = null)
             {
-                var regex = GetRegexForMatch(matchText);
-                return regex.Replace(replaceInsideString, replacementText);
+                if (subSpansToReplace == null)
+                {
+                    // We do not have already computed sub-spans to replace inside the string.
+                    // Get regex for matches within the string and replace all matches with replacementText.
+                    var regex = GetRegexForMatch(matchText);
+                    return regex.Replace(replaceInsideString, replacementText);
+                }
+                else
+                {
+                    // We are provided specific matches to replace inside the string.
+                    // Process the input string from start to end, replacing matchText with replacementText
+                    // at the provided sub-spans within the string for these matches.
+                    var stringBuilder = new StringBuilder();
+                    var startOffset = 0;
+                    foreach (var subSpan in subSpansToReplace)
```

Comment:
i feel like there's gotta be an easier way to do this.  but i'm too dumb right now to think of it.

Reply:
Yes, potentially, but the current logic is not that complex so I think it is fine.